.github/workflows/conda_ci_container_cicd.yml

```diff
@@ -0,0 +1,46 @@
+name: Build and Publish Conda CI Images
+on:
+  schedule:
+    - cron:  '0 0 * * 0'
+  pull_request:
+    paths:
+      - 'ci/Dockerfile'
+      - '.github/workflows/conda_ci_container_cicd.yml'
+  push:
+    paths:
+      - 'ci/Dockerfile'
+      - '.github/workflows/conda_ci_container_cicd.yml'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.7, 3.8, 3.9]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        if: github.ref == 'refs/heads/master'
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          version: latest
+          driver-opts: network=host
+
+      - name: build-push conda-ci-linux-64-python-${{ matrix.python }}
+        uses: docker/build-push-action@v2
+        with:
+          context: ./ci
+          builder: ${{ steps.buildx.outputs.name }}
+          file: ./ci/Dockerfile
+          build-args: PYTHON_VERSION=${{ matrix.python }}
+          tags: conda/conda-ci-linux-64:python${{ matrix.python }}-latest
+          push: ${{ github.ref == 'refs/heads/master' }}
```

Comment:
The login step above has the same condition but the buildx step doesn't. Is this intentional?

Could we move this condition to the whole job so we don't even build the image?

Reply:
Login only works if credentials are available... they typically are restricted to master... but the workflow can still run on other branches without login/push to test building the images if there are updates.

Reply:
Got it, thanks.